packaging/conda/build_conda.py

```diff
@@ -0,0 +1,166 @@
+        os.environ["SOURCE_ROOT_DIR"] = SOURCE_ROOT_DIR
+        os.environ["CONDA_CUDATOOLKIT_CONSTRAINT"] = version_constraint(self.cuda_version)
+        os.environ["FORCE_CUDA"] = "1"
+
+        if self.conda_always_copy:
+            os.environ["CONDA_ALWAYS_COPY"] = "true"
+
+
+    def _get_build_args(self):
+        args = [
+            "conda",
+            "build",
+            "-c",
+            "fastchan",  # which can avoid needing pytorch and conda-forge
+            "--no-anaconda-upload",
+            "--python",
+            self.python_version,
+        ]
+        if self.conda_debug:
+            args += ["--debug"]
+        args += ["--dirty"]
+        args += ["--croot", "../build"]
+        return args + ["packaging/conda/xformers"]
+
+    def do_build(self):
+        self._set_env_for_build()
+        args = self._get_build_args()
+        print(args)
+        subprocess.check_call(args)
+
+    def build_in_docker(self):
+        filesystem = subprocess.check_output("stat -f -c %T .", shell=True)
+        if filesystem in (b"nfs", b"tmpfs"):
+            raise ValueError("Cannot run docker here. " + "Please work on a local filesystem, e.g. /raid.")
+        image = conda_docker_image_for_cuda(self.cuda_version)
+        args = ["sudo", "docker", "run", "-it", "--rm", "-w", "/m"]
+        args += ["-v", f"{SOURCE_ROOT_DIR}:/m", image]
+        args += ["python3", str(THIS_PATH.relative_to(SOURCE_ROOT_DIR))]
+        self_args = [
+            "--cuda",
+            self.cuda_version,
+            "--pytorch",
+            self.pytorch_version,
+            "--python",
+            self.python_version,
+        ]
+        args += self_args
+        print(args)
+        subprocess.check_call(args)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Build the conda package.")
+    parser.add_argument(
+        "--python", metavar="3.X", required=True, help="python version e.g. 3.10"
+    )
+    parser.add_argument(
+        "--cuda", metavar="1X.Y", required=True, help="cuda version e.g. 11.3"
+    )
+    parser.add_argument(
+        "--pytorch", metavar="1.Y.Z", required=True, help="PyTorch version e.g. 1.11.0"
+    )
+    parser.add_argument(
+        "--docker", action="store_true", help="Call this script inside docker."
+    )
+
+    args = parser.parse_args()
+
+    pkg = Build(
+        python_version=args.python, pytorch_version=args.pytorch, cuda_version=args.cuda
+    )
+
+    if args.docker:
+        pkg.build_in_docker()
+    else:
+        pkg.do_build()
+
+
+# python packaging/conda/build_conda.py  --cuda 11.6 --python 3.10 --pytorch 1.12.1
+# python packaging/conda/build_conda.py  --cuda 11.3 --python 3.9 --pytorch 1.12.1  # <= the dino one
+# python packaging/conda/build_conda.py  --cuda 11.6 --python 3.10 --pytorch 1.11.0
+
+# Note this does the build outside the root of the tree.
+
+# TODO:
+# - Make a local conda package cache available inside docker
+# - use ninja
```

Comment:
this is going to be important for parallelizing the builds. I'd hope it should be as simple as getting `ninja` installed through conda beforehand